### PR TITLE
[Graphbolt] Support feature metadata.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/feature_store.py [hunk 1]
```diff
@@ -52,6 +52,16 @@ def update(self, value: torch.Tensor, ids: torch.Tensor = None):
         """
         raise NotImplementedError
 
+    def metadata(self):
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/feature_store.py:55]: existing `.size()` is one of the metadata? Or a special metadata that lies standalone?

File: python/dgl/graphbolt/feature_store.py [hunk 2]
```diff
@@ -110,6 +120,29 @@ def size(
         """
         raise NotImplementedError
 
+    def metadata(
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/feature_store.py:123]: do we need to think more about the `metadata`? In this implementation, metadata is always per triplet of `domain/type_name/feature_name`. do need some metadata that's per `domain`, or even per `feature_store`?

File: python/dgl/graphbolt/feature_store.py (hunk 1 above)
Comment (frozenbugs) [reply python/dgl/graphbolt/feature_store.py:55]: Size is actually inferred from the tensor.

File: python/dgl/graphbolt/feature_store.py (hunk 2 above)
Comment (frozenbugs) [reply python/dgl/graphbolt/feature_store.py:123]: Good point, maybe in the future the API can be extended to making domain / type / feature optional.

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**